### DISCUSSION TITLE
Vercel Edge Config PATCH always responds with "The request is missing an authentication token"

### DISCUSSION THREAD

**Discussion by sandervspl** [vercel/community#1177] [Help] [answered]
I'm trying to use the PATCH endpoint `https://api.vercel.com/v1/edge-config/<MY_PROJECT_ID>/items` but I can't get it to work, even with the example curl.

<img width="723" alt="Screenshot 2022-12-20 at 17 15 25" src="https://user-images.githubusercontent.com/4492636/208717034-2b1ff7ca-b6af-4272-80f1-b79212c84d8b.png">
<img width="751" alt="Screenshot 2022-12-20 at 17 15 54" src="https://user-images.githubusercontent.com/4492636/208717041-59579e0b-dd2e-47c3-9c4a-537811e86577.png">

My guess is that I need an access token that allows `write` but I can not figure out how to generate such a token.

**Comment by amyegan** [comment]
Sorry you're running into this error @sandervspl. This might seem like a silly question, but did you already [create your Edge Config](https://vercel.com/docs/concepts/edge-network/edge-config/edge-config-api#create-an-edge-config) using the same auth token? It looks like you must have since you have an edge config id, but I don't want to make assumptions about what is working and what is not.

**Reply by sandervspl** [reply]
Yes I have! All GET requests are working with the same access token

**Reply by amyegan** [reply]
Edge Config creation should be a `POST`. Did that work for you as well?

**Reply by sandervspl** [reply]
I'm following this example to update a key with a new value https://vercel.com/docs/concepts/edge-network/edge-config/edge-config-api#update-your-edge-config

**Comment by dferber90** [comment, marked as answer]
In the screenshot it looks like you're sending the Edge Config's read-access-token as the Authorization header, but this endpoint requires an Vercel API token. You should be able to use the same one you used when creating the Edge Config. Does that work?

Basically
- The Edge Config's read-access-token is only used when talking to `edge-config.vercel.com`
- The Vercel API token is used when talking to `api.vercel.com`

**Reply by sandervspl** [reply]
Thanks, the Vercel access token did the trick! That I needed a different token for these requests was not immediately clear for me from reading the docs. My suggestion would be to change the placeholders so they are different from each other (`your_vercel_token_here` and `your_edge_config_token_here` or something)
